Significant changes to dashboard

joeyfreeman opened the pull request:
Changed dashboard design:
-Reduced number of pages to 4 basic layouts: welcome, navigation (TV
show -> seasons filtering), account, and media item
-Changed welcome page to feature one textbox. Added example text.
-Implemented basic search functionality by show name. Search is
case-insensitive.
-Removed account details page and fixed two window issue
-Created basic media item page
Other changes:
-Added default constructor to Season
-Changed TvShow to contain vector of Seasons
-Changed seasonNumber in TvShow to be int, not QString
-Removed numberOfEpisodes() method from TvShow - can just call size() on
episodes data member
-Modified testseason to reflect changes

joeyfreeman committed:
Significant changes to dashboard

Changed dashboard design:
-Reduced number of pages to 4 basic layouts: welcome, navigation (TV
show -> seasons filtering), account, and media item
-Changed welcome page to feature one textbox. Added example text.
-Implemented basic search functionality by show name. Search is
case-insensitive.
-Removed account details page and fixed two window issue
-Created basic media item page
Other changes:
-Added default constructor to Season
-Changed TvShow to contain vector of Seasons
-Changed seasonNumber in TvShow to be int, not QString
-Removed numberOfEpisodes() method from TvShow - can just call size() on
episodes data member
-Modified testseason to reflect changes

haleyminshell commented:
That looks really awesome! I really like it! 
2 things I noticed though:
- For the media details page, I think it would be nice if it had a frame behind the text so it's not directly on the blue background. 
- It still needs the page before going directly to the media information page that would allow the user to check off a box saying they want to add a show as watched and will not display episode details/comments.

joeyfreeman committed:
Added spoiler alert and background to media page

Watched episodes are not saved yet because user class is not complete

joeyfreeman commented:
I added a background and a temporary spoiler alert. For now, the alert pops up every time you go to an episode because the user/person class does not have a watched episode list yet.

ErickHeyl reviewed (commented):
TreeWidgets on dash looks good and seems to be a good start to populate list

joeyfreeman committed:
Added logout button and episode summary field

Also changed account details page to include first and last names and
allow password changes

joeyfreeman committed:
fix merge conflict
